### DISCUSSION TITLE
What event fires if there is any change

### DISCUSSION THREAD

**Discussion by questsin** [vakata/jstree#2537] [Q&A]
When working with the context menu, All the below code seems to fire prior to the rename, not after it. If the Tree changes, I'd like to save it to local storage, not just the state of open/closed and checked

        $('#jstree_demo_div').on("changed.jstree", function (e, data) {
            console.log('changed');
        });
        $('#jstree_demo_div').on("create_node.jstree,set_text.jstree,rename_node.jstree,move_node.jstree,delete_node.jstree", function (e, data) {
            console.log('crud');
        })    
        $('#jstree_demo_div').on("model.jstree", function (e, data) {
            console.log('model');
        })

**Comment by vakata** [comment]
All those events happen AFTER the change has occurred. Just in case - attach the handlers after creating the tree, so that internal handlers have priority. You could also debounce your handler (give it some time after the event and before saving to localStorage).
